torch/distributed/_composable/fsdp/_fsdp_param.py

```diff
@@ -199,21 +200,48 @@ def _init_sharded_param(self, param: nn.Parameter, device: torch.device):
                     "FSDP requires the DP and TP mesh to have the same parent mesh but got: \n"
                     f"DP's global mesh: {dp_global_mesh}\nTP's global mesh: {tp_global_mesh}"
                 )
-            self._global_mesh = dp_global_mesh
             if len(self._tp_spec.placements) != 1:
                 raise NotImplementedError(
                     f"FSDP only supports 1D TP, not {self._tp_spec.placements}"
                 )
             global_placements: List[Placement] = [Replicate(), Replicate()]
             global_dp_mesh_dim = _mesh_resources.get_parent_mesh_dim(dp_mesh)
             global_tp_mesh_dim = _mesh_resources.get_parent_mesh_dim(tp_mesh)
+
+            def infer_spmd_mesh(global_mesh, global_dp_mesh_dim, global_tp_mesh_dim):
+                from torch.distributed.device_mesh import _mesh_resources, DeviceMesh
+                pp_submesh_tensors = global_mesh.mesh.unbind(0)
```

Comment:
what does this do?

Reply:
it's basically generate from a 3-D mesh to a list of 2D meshes. i.e. 
```
>>> import torch
>>> a = torch.arange(8).reshape(2, 2, 2)
>>> a
tensor([[[0, 1],
         [2, 3]],

        [[4, 5],
         [6, 7]]])
>>> a.unbind(0)
(tensor([[0, 1],
        [2, 3]]), tensor([[4, 5],
        [6, 7]]))
```